doc/examples.md

```diff
@@ -0,0 +1,210 @@
+  "systemd": {
+    "units": [{
+      "name": "example.service",
+      "enable": true,
+      "contents": "[Service]\nType=oneshot\nExecStart=/usr/bin/echo Hello World\n\n[Install]\nWantedBy=multi-user.target"
+    }]
+  }
+}
+```
+
+### example.service
+
+```
+[Service]
+Type=oneshot
+ExecStart=/usr/bin/echo Hello World
+
+[Install]
+WantedBy=multi-user.target
+```
+
+## Reformat the Root Filesystem
+
+This example Ignition configuration will locate the device with the "ROOT" filesystem label (the root filesystem) and reformat it to btrfs, recreating the filesystem label. The `force` option is set to ensure that `mkfs.btrfs` ignores any existing filesystem.
+
+### Btrfs
+
+```json
+{
+  "ignition": { "version": "2.0.0" },
+  "storage": {
+    "filesystems": [{
+      "device": "/dev/disk/by-label/ROOT",
+      "format": "btrfs",
+      "create": {
+        "force": true,
+        "options": [ "--label=ROOT" ]
+      }
+    }]
+  }
+}
+```
+
+### XFS
+
+```json
+{
+  "ignition": { "version": "2.0.0" },
+  "storage": {
+    "filesystems": [{
+      "device": "/dev/disk/by-label/ROOT",
+      "format": "xfs",
+      "create": {
+        "force": true,
+        "options": [ "-L", "ROOT" ]
+      }
+    }]
+  }
+}
+```
+
+The create options are forwarded to the underlying `mkfs.$format` utility. The respective `mkfs.$format` manual pages document the available options.
+
+## Create Files on the Root Filesystem
+
+In many cases it is useful to write files to the root filesystem. This example writes a single file to `/foo/bar` on the root filesystem. The contents of the file ("example file") are specified inline in the config using the [data URL scheme][rfc2397].
+
+```json
+{
+  "ignition": { "version": "2.0.0" },
+  "storage": {
+    "files": [{
+      "filesystem": "root",
+      "path": "/foo/bar",
+      "contents": { "source": "data:,example%20file%0A" }
+    }]
+  }
+}
+```
+
+The config makes use of the universally-defined "root" filesystem. This filesystem is defined within Ignition itself and roughly looks like the following. The "root" filesystem allows additional configs to reference the root filesystem, regardless of its type (e.g. btrfs, tmpfs, ext4).
+
+```json
+{
+  "ignition": { "version": "2.0.0" },
+  "storage": {
+    "filesystems": [{
+      "name": "root",
+      "path": "/sysroot"
+    }]
+  }
+}
+```
+
+## Create Files from Remote Contents
+
+There are cases where it is desirable to write a file to disk, but with the contents of a remote resource. The following config demonstrates how to do this in addition to validating the contents of the file.
+
+```json
+{
+  "ignition": { "version": "2.0.0" },
+  "storage": {
+    "files": [{
+      "filesystem": "root",
+      "path": "/foo/bar",
+      "contents": {
+        "source": "http://example.com/asset",
+        "verification": { "hash": "sha512-0123456789abcdef0123456789abcdef0123456789abcdef0123456789abcdef0123456789abcdef0123456789abcdef0123456789abcdef0123456789abcdef" }
```

Comment:
Is whitespace trimming being done before hashing. Might be useful to show the command line equivalent for getting the right hash. Saves people a cycle of getting it wrong and updating.